x-pack/filebeat/input/http_endpoint/crc.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package http_endpoint
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// Type Validator points to the function in charge of validating
+// CRC requests defined for each provider in the crcValidator struct.
+//
+// Params:
+//
+//	*crcValidator: pointer to the CRC handler for the webhook provider
+//	mapstr.M: input JSON to be validated
+//
+// Returns:
+//
+//	status: response code to send back to the endpoint
+//	resp: response body
+//	error: failure reason if proceed
+type Validator func(*crcValidator, mapstr.M) (status int, resp string, _ error)
```

Comment:
Go documentation tends towards less structured text and more natural language. There is a common set of idioms that are outlined [here](https://go.dev/doc/comment). There are broader Go review guidelines [here](https://github.com/golang/go/wiki/CodeReviewComments) and for testing [here](https://github.com/golang/go/wiki/TestComments) (though note that assertion libraries are used by many people here, so that is localised).

Here is how I'd write this.
```suggestion
// Validator is a CRC validation function type. It applies the provided validator
// to the challenge sent by an API and returns an HTTP status code and response
// body to be sent to the challenging API server.
type Validator func(validator *crcValidator, challenge mapstr.M) (status int, resp string, _ error)
```
Errors are not usually explicitly mentioned unless the type/value of the error is going to be useful as part of the contract, for example [here](https://pkg.go.dev/os#Stat).

Reply:
@efd6, thanks for the suggestion and the documentation. Already updated.